### ISSUE TITLE
SelectList and EventCallback

### ISSUE THREAD

**Issue by stevenlupo** [stsrki/Blazorise#1336]
The following code causes a compilation error for `SelectedValueChanged`.
```
<SelectList Data="@myDdlData"
   TextField="@((item)=>item.MyTextField)"
   ValueField="@((item)=>item.MyValueField)"
   SelectedValue="@selectedListValue"
   SelectedValueChanged="@MyListValueChangedHandler" />

@code{
    void MyListValueChangedHandler(object newValue)
    {
        selectedListValue = newValue;
    }
}
```

The error:

`CS1503	Argument 12: cannot convert from 'Microsoft.AspNetCore.Components.EventCallback' to 'Microsoft.AspNetCore.Components.EventCallback<object>'`

**Comment by stsrki** [comment]
What version you're using? In one of the latest preview's we introduced generic SelectList and now you need to define a `TValue` for the item value.

eg.

```
<SelectList Data="@myDdlData"
   TValue="int"
   TextField="@((item)=>item.MyTextField)"
   ValueField="@((item)=>item.MyValueField)"
   SelectedValue="@selectedListValue"
   SelectedValueChanged="@MyListValueChangedHandler" />

@code{
    void MyListValueChangedHandler(int newValue)
    {
        selectedListValue = newValue;
    }
}
```

**Comment by stevenlupo** [comment]
I'm using 0.9.2-preview8.

Adding the `TValue `does not work, it then wants `TItem `and that results in more compile errors.

If that code looks familiar, it's from the Blazorise Demo:
https://github.com/stsrki/Blazorise/blob/master/Demos/Blazorise.Demo/Pages/Tests/ComponentsPage.razor

**Comment by stsrki** [comment]
I was quick to write an example without testing. Sorry about that. You should also add a `TItem` type param.

```
<SelectList TItem="MySelectModel"
            TValue="int"
            Data="@myDdlData"
            TextField="@((item)=>item.MyTextField)"
            ValueField="@((item)=>item.MyValueField)"
            SelectedValue="@selectedintValue"
            SelectedValueChanged="@MyIntValueChangedHandler" />

@code{
    int selectedintValue { get; set; } = 3;

    void MyIntValueChangedHandler( int newValue )
    {
        selectedintValue = newValue;
    }
}
```

Also, the documentation that you see in `master` branch is valid for 0.9.1 only, The new documentation will be published once the 0.9.2 is released.

**Comment by stevenlupo** [comment]
That worked, thanks.
